R/reactable.R

```diff
@@ -809,6 +807,86 @@ renderReactable <- function(expr, env = parent.frame(), quoted = FALSE) {
   htmlwidgets::shinyRenderWidget(expr, reactableOutput, env, quoted = TRUE)
 }
 
+#' Convert a reactable widget to HTML tags
+#'
+#' This S3 method exists to enable [reactable()]'s `static` rendering option.
+#'
+#' @param x a [reactable()] instance.
+#' @param standalone
+#'
+#' @export
+as.tags.reactable <- function(x, standalone = FALSE) {
+  # Should call htmlwidgets:::as.tags.htmlwidget(), which calls htmlwidgets::toHTML()
+  result <- NextMethod("as.tags", x, standalone = standalone)
+
+  # Make sure react is loaded before the reactable dependencies
+  dependencies <- c(
+    list(
+      reactR::html_dependency_corejs(), # Necessary for RStudio Viewer version < 1.2 and IE11
+      reactR::html_dependency_react(),
+      reactR::html_dependency_reacttools()
+    ),
+    htmlDependencies(result)
+  )
```

Comment:
For some reason, these dependencies aren't coming through when rendering tables in Shiny outputs or R Notebooks. R Notebooks might be an unrelated issue with the IDE (https://github.com/glin/reactable/issues/43), but for Shiny, it seems like `as.tags.reactable()`/`as.tags()` is never called? If that's how it works for htmlwidgets in Shiny, could we move the html dependencies back to widget_html? That would probably fix the R Notebook issue too.

Reply:
Ah, yes, good point!